plugins/meshy/meshy.js

```diff
@@ -0,0 +1,1236 @@
+    }
+    return [
+        normalSum[0] / normalLength,
+        normalSum[1] / normalLength,
+        normalSum[2] / normalLength
+    ];
+}
+
+function multiplyScalar(vec, scalar) {
+    return vec.map((coord) => coord * scalar);
+}
+
+function rotatePoint(point, center, rotation) {
+    // Convert rotation angles to radians
+    const [rx, ry, rz] = rotation.map(Math.degToRad);
+
+    // Translate point to origin
+    let [x, y, z] = point.map((coord, i) => coord - center[i]);
+
+    // Rotate around X-axis
+    let temp = y;
+    y = y * Math.cos(rx) - z * Math.sin(rx);
+    z = temp * Math.sin(rx) + z * Math.cos(rx);
+
+    // Rotate around Y-axis
+    temp = x;
+    x = x * Math.cos(ry) + z * Math.sin(ry);
+    z = -temp * Math.sin(ry) + z * Math.cos(ry);
+
+    // Rotate around Z-axis
+    temp = x;
+    x = x * Math.cos(rz) - y * Math.sin(rz);
+    y = temp * Math.sin(rz) + y * Math.cos(rz);
+
+    // Translate back
+    return [
+        x + center[0],
+        y + center[1],
+        z + center[2]
+    ];
+}
+(function() {
+const pluginInfo = {"name":"Meshy","id":"meshy","version":"1.0.2","repository":"https://github.com/Shadowkitten47/Meshy"};
+
+if (!settings["normalized_uvs"])
+    new Setting("normalized_uvs", {
+        name: "Normalize UVs",
+        description: "Normalize uvs on export",
+        value: true,
+        plugin: pluginInfo.id
+    })
+if (!settings["meta_data"])
+    new Setting("meta_data", {
+        name: "Meta Data",
+        description: "Adds meta data to mesh. ( For smaller file size disable this )",
+        value: true,
+        plugin: pluginInfo.id
+    })
+if (!settings["skip_normals"]) {
+    new Setting("skip_normals", {
+        name: "Skip Normals",
+        description: "Model will lack all shading information",
+        value: false,
+        plugin: pluginInfo.id
+    })
+}
+if (!settings["Force Multi-Textures"])
+    new Setting("force_textures", {
+        name: "Force Multi-Textures",
+        description: "Forces bedrock formats to use Multi-Textures ( You will need to stitch the textures )",
+        value: false,
+        plugin: pluginInfo.id,
+        onChange: (value) => {
+            Formats['bedrock'].single_texture = !value
+            Formats['bedrock_old'].single_texture = !value
+        }
+})
+
+Plugin.register(pluginInfo.id, {
+	title: pluginInfo.name,
+	author: 'Shadowkitten47',
+	icon: 'diamond',
+	description: 'Enables the use of a meshes in bedrock formats and to export them to Minecraft Bedrock',
+	version: pluginInfo.version,
+	variant: 'both',
+    repository: pluginInfo.repository,
+    onload() {
+        const bedrock_old = Formats['bedrock_old']
+        const bedrock = Formats['bedrock']
+        bedrock.meshes = true;
+        bedrock_old.meshes = true;
+        bedrock.single_texture = !settings["single_texture"]?.value
+        bedrock_old.single_texture = !settings["single_texture"]?.value
+       
+    },
+    onunload() {
```

Comment:
Settings and actions are not cleaned up when uninstalling. You can do this here via setting_xy.delete(); etc.